### PR TITLE
VoyageAI reranking

### PR DESCRIPTION
**Pull request by fzowl**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Version Bump?

Did I bump the version in the `pyproject.toml` file of the package I am updating? (Except for the `llama-index-core` package)

- [ ] Yes
- [ ] No

## Type of Change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Added new notebook (that tests end-to-end)

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

File: docs/examples/pipeline/query_pipeline_voyageai.ipynb [hunk 1]
```diff
@@ -0,0 +1,357 @@
+{
```
Comment (hatianzhang) [inline docs/examples/pipeline/query_pipeline_voyageai.ipynb:1]: <p>is this the right example for reranking?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/12111/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (fzowl) [reply docs/examples/pipeline/query_pipeline_voyageai.ipynb]: I removed the notebook file, i used the same for testing purposes.

**Comment by hatianzhang** [comment]
@fzowl could you help fix the unit testing

**Comment by fzowl** [comment]
@hatianzhang I believe i corrected the problem.

File: llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+[tool.mypy]
+disallow_untyped_defs = true
+# Remove venv skip when integrated with pre-commit
+exclude = ["_static", "build", "examples", "notebooks", "venv"]
+ignore_missing_imports = true
+python_version = "3.8"
+
+[tool.poetry]
+authors = ["Your Name <you@example.com>"]
```
Comment (hatianzhang) [inline llama-index-integrations/postprocessor/llama-index-postprocessor-voyageai-rerank/pyproject.toml:20]: @fzowl can you add your name or org here

**Review by hatianzhang — APPROVED** [review]

**Pull request merged**